src/Adapter/Product/QueryHandler/GetProductForEditingHandler.php

```diff
@@ -169,14 +180,24 @@ private function getBasicInformation(Product $product): ProductBasicInformation
     /**
      * @param Product $product
      *
-     * @return ProductCategoriesInformation
+     * @return CategoriesInformation
      */
-    private function getCategoriesInformation(Product $product): ProductCategoriesInformation
+    private function getCategoriesInformation(Product $product): CategoriesInformation
     {
-        $categoryIds = array_map('intval', $product->getCategories());
+        $categoryIds = $product->getCategories();
         $defaultCategoryId = (int) $product->id_category_default;
 
-        return new ProductCategoriesInformation($categoryIds, $defaultCategoryId);
+        $categoriesInformation = [];
+        foreach ($categoryIds as $categoryId) {
```

Comment:
It may be a bit heavy to get each product's categories just to get their name It might be worth it to add a function in the repository to get multiple names by list of IDs in a single request

Reply:
Again I think a dedicated method in the repository might be useful Some products have dozens of categories assigned to them which would drop the performance of the page that's what we try to avoid with this new version